### PR TITLE
DNID-350 content backlinks

### PR DESCRIPTION
**Pull request by johangant**
Early PR. A few minor things left to sort out, but broadly:

- Need to refactor to remove `nidirect_` prefixes to make it more friendly for release as contrib module.

### CODE REVIEW

File: nidirect_backlinks/src/Command/ScanCommand.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    if (!$db_conn->schema()->tableExists('nidirect_backlinks')) {
+      return 3;
+    }
+
+    $query = $db_conn->query('select nid from {node}');
+    $results = $query->fetchAll();
+
+    // Potentially long running task here; may need batching.
+    foreach ($results as $row) {
+      $node = \Drupal::service('entity_type.manager')->getStorage('node')->load($row->nid);
```
Comment (omahm) [inline nidirect_backlinks/src/Command/ScanCommand.php]: You're gonna be calling getStorage() on every iteration, I'd look at assigning this outside the loop and possibly injecting it as a dependency.

File: nidirect_backlinks/src/Command/ScanCommand.php [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+      if ($node instanceof NodeInterface) {
+        \Drupal::service('nidirect_backlinks.linkmanager')->processEntity($node);
+        // $this->getIo()->info('Processed node ID: ' . $row->nid);
```
Comment (omahm) [inline nidirect_backlinks/src/Command/ScanCommand.php]: Remove commented out code.

File: nidirect_backlinks/nidirect_backlinks.module [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      return $output;
+
+    default:
+  }
+}
+
+/**
+ * Implements hook_node_insert().
+ */
+function nidirect_backlinks_node_insert(EntityInterface $entity) {
```
Comment (omahm) [inline nidirect_backlinks/nidirect_backlinks.module]: Is there an event you can subscribe to for these, or can it only be done with hooks?

**Comment by johangant** [comment]
Good comments, and now reflected in the repo at https://git.drupalcode.org/project/whatlinkshere

I'll close here as this PR will become a composer and config change to get this working.

**Pull request closed without merging**